lib/Doctrine/Common/Annotations/CachedReader.php

```diff
@@ -178,14 +198,14 @@ public function clearLoadedAnnotations()
     /**
      * Fetches a value from the cache.
      *
-     * @param string          $cacheKey The cache key.
-     * @param ReflectionClass $class    The related class.
+     * @param string $cacheKey The cache key.
      *
      * @return mixed The cached value or false when the value is not in cache.
      */
     private function fetchFromCache($cacheKey, ReflectionClass $class)
     {
-        if (($data = $this->cache->fetch($cacheKey)) !== false) {
+        $cacheKey = strtr($cacheKey, '\\', '.');
```

Comment:
should this be done when generating the cache key rather than altering it here ? I find it confusing that some methods (like `isCacheFresh`) assume a valid PSR-16 cache key while others use the invalid cache key, with the same variable names. Making sure that we have valid PSR-16 cache keys everywhere in variables named `$cacheKey` seems better to me (and AFAICT, we only have 2 places generating them, so that won't duplicate the logic more than your current approach)

Reply:
Good point, will do 👍